blocks/model.py

```diff
@@ -161,15 +162,18 @@ def __init__(self, outputs):
             raise ValueError("top bricks with the same name:"
                              " {}".format(', '.join(repeated_names)))
 
-        brick_param_names = {
-            v: k for k, v in Selector(self.top_bricks).get_params().items()}
-        self.params = []
-        for param in VariableFilter(roles=[PARAMETER])(self.shared_variables):
-            if param in brick_param_names:
-                self.params.append((brick_param_names[param], param))
+        brick_parameter_names = {
+            v: k
+            for k, v in Selector(self.top_bricks).get_parameters().items()}
+        self.parameters = []
```

Comment:
Indeed, a situation when a class has `parameters` property, `get_params` method and `params` attribute is rather confusing!

The real solution would be a brutal refactoring of the `Model` class. But to finish this PR I would suggest

- rename `params` attribute to `_params`, that is simply hide it from user eyes

-  rename `get_params` into `get_parameter_dict`. As far as I remember this method is used in the `reverse_words` demo in `blocks-examples`, which means that it should be also renamed there

I think that `ComputationGraph.parameters` should be kept, since it a popular property with very clear semantics.

Does it make sense?